packages/apps/plugins/plugin-sketch/src/components/Sketch.tsx

```diff
@@ -30,83 +25,116 @@ export type SketchComponentProps = {
-        x: (width - commonBounds.width * zoom) / 2 / zoom - commonBounds.minX,
-        y: (height - commonBounds.height * zoom) / 2 / zoom - commonBounds.minY,
-      };
-      editor.animateCamera(center.x, center.y, zoom, animate ? { duration: 250 } : undefined);
-      setReady(true);
+  useEffect(() => {
+    if (!editor || !adapter || width === undefined || height === undefined) {
+      return;
     }
-  };
 
-  useEffect(() => {
-    if (!autoZoom || width < 1 || (height ?? 0) < 1) {
+    editor.zoomToFit({ duration: 0 });
+    editor.resetZoom();
+
+    setReady(true);
+    if (!autoZoom) {
       return;
     }
 
-    editor?.updateViewportScreenBounds();
-    zoomToContent(false);
+    const zoom = (animate = true) => {
+      zoomToContent(editor, width, height, maxZoom, animate);
+    };
 
-    const onUpdate = debounce(zoomToContent, 200);
-    const subscription = store.listen(() => onUpdate(true), { scope: 'document' });
-    return () => subscription();
-  }, [editor, width, height]);
+    zoom(false);
+    const onUpdate = debounce(zoom, 200);
+    const subscription = readonly ? adapter.store!.listen(() => onUpdate(true), { scope: 'document' }) : undefined;
+    return () => subscription?.();
+  }, [editor, adapter, width, height, autoZoom]);
 
-  // https://github.com/tldraw/tldraw/blob/main/packages/ui/src/lib/TldrawUi.tsx
-  // TODO(burdon): Customize assets: https://tldraw.dev/docs/assets
+  if (!adapter) {
+    return null;
+  }
+
+  // https://tldraw.dev/docs/user-interface
   return (
     <div
       ref={containerRef}
       style={{ visibility: ready ? 'visible' : 'hidden' }}
-      className={mx(
-        'is-full bs-full',
-        // 14Override z-index.
-        '[&>div>span>div]:z-0',
-        // 14Hide .tlui-menu-zone
-        '[&>div>main>div:nth-child(1)>div:nth-child(1)]:hidden',
-        // 14Hide .tlui-navigation-zone
-        '[&>div>main>div:nth-child(2)>div:nth-child(1)>div:nth-child(1)]:hidden',
-        // 14Hide .tlui-help-menu
-        '[&>div>main>div:nth-child(2)>div:nth-child(1)>div:nth-child(3)]:hidden',
-        // 14Hide .tlui-debug-panel
-        '[&>div>main>div:nth-child(2)>div:nth-child(2)]:hidden',
-        styles.background,
-        className,
-      )}
+      className={mx('is-full bs-full', className)}
+      onPointerEnter={() => {
+        setActive(!readonly && !adapter.readonly);
+      }}
+      onPointerLeave={() => {
+        setActive(false);
+      }}
     >
       {/* NOTE: Key forces unmount; otherwise throws error. */}
-      <Tldraw key={sketch.id} store={store} hideUi={readonly} onMount={setEditor} />
+      <Tldraw
+        // Setting the key forces re-rendering when the content changes.
+        key={sketch.id}
+        store={adapter.store}
+        hideUi={!active}
+        // TODO(burdon): Customize assets: https://tldraw.dev/docs/assets
+        maxAssetSize={1024 * 1024}
+        components={{
+          DebugPanel: null,
+          Grid: CustomGrid,
```

Comment:
What if we don't like this new grid and prefer the TLDraw-standard grid? Can this be a setting?